sentry/src/main/java/io/sentry/TransactionPerformanceCollector.java

```diff
@@ -32,64 +30,83 @@ public TransactionPerformanceCollector(final @NotNull SentryOptions options) {
       options
           .getLogger()
           .log(
               SentryLevel.INFO,
-              "Memory collector is a NoOpMemoryCollector. Memory stats will not be captured during transactions.");
+              "Memory collector is a NoOpCollector. Memory stats will not be captured during transactions.");
+    }
+    if (isCpuCollectorNoOp) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.INFO,
+              "Cpu collector is a NoOpCollector. Cpu stats will not be captured during transactions.");
+    }
+    if (isMemoryCollectorNoOp && isCpuCollectorNoOp) {
       return;
     }
-    if (!memoryMap.containsKey(transaction.getEventId().toString())) {
-      memoryMap.put(transaction.getEventId().toString(), new ArrayList<>());
+
+    if (!performanceDataMap.containsKey(transaction.getEventId().toString())) {
+      performanceDataMap.put(transaction.getEventId().toString(), new PerformanceCollectionData());
       options
           .getExecutorService()
           .schedule(
               () -> {
-                ArrayList<MemoryCollectionData> memoryCollectionData =
-                    (ArrayList<MemoryCollectionData>) stop(transaction);
-                if (memoryCollectionData != null) {
-                  memoryMap.put(transaction.getEventId().toString(), memoryCollectionData);
+                PerformanceCollectionData data = stop(transaction);
+                if (data != null) {
+                  performanceDataMap.put(transaction.getEventId().toString(), data);
                 }
               },
               TRANSACTION_COLLECTION_TIMEOUT_MILLIS);
     }
     if (!isStarted.getAndSet(true)) {
       synchronized (timerLock) {
+        cpuCollector.setup();
         if (timer == null) {
           timer = new Timer(true);
         }
         timer.scheduleAtFixedRate(
             new TimerTask() {
               @Override
               public void run() {
+                MemoryCollectionData memoryData = null;
                 if (memoryCollector != null) {
-                  MemoryCollectionData memoryData = memoryCollector.collect();
-                  if (memoryData != null) {
-                    synchronized (timerLock) {
-                      for (ArrayList<MemoryCollectionData> list : memoryMap.values()) {
-                        list.add(memoryData);
-                      }
-                    }
+                  memoryData = memoryCollector.collect();
+                }
+                CpuCollectionData cpuData = null;
+                if (cpuCollector != null) {
+                  cpuData = cpuCollector.collect();
+                }
+                synchronized (timerLock) {
+                  for (PerformanceCollectionData data : performanceDataMap.values()) {
+                    data.addData(memoryData, cpuData);
                   }
                 }
               }
             },
-            0,
+            TRANSACTION_COLLECTION_INTERVAL_MILLIS,
```

Comment:
`m`: any reason you're providing a delay parameter here?

Reply:
so we let some time pass between setup() and collect() calls.
This way ICollectors that collect average stats based on time intervals, like AndroidCpuCollector, can have an actual time interval to evaluate.
Just added a comment to clarify it